python/cudf/cudf/core/column/column.py

```diff
@@ -1570,9 +1574,12 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
         if is_categorical_dtype(arbitrary.dtype):
             arb_dtype = arbitrary.dtype
         else:
-            arb_dtype = check_cast_unsupported_dtype(arbitrary.dtype)
-            if arb_dtype != arbitrary.dtype.numpy_dtype:
-                arbitrary = arbitrary.astype(arb_dtype)
+            if arbitrary.dtype == pd.StringDtype():
+                arb_dtype = np.dtype("O")
```

Comment:
Do we want to wait for a follow up PR to move from `object` dtype to a proper `String` dtype?

Reply:
Yeah, I had planned to replace `object` with either an extension type or a `cudf.Dtype` in a PR that handles it everywhere in the codebase.

Reply:
👍

Reply:
Maybe file an issue if one doesn't already exist and add a #TODO comment there?

Reply:
created issue https://github.com/rapidsai/cudf/issues/5714.